### PR TITLE
Delete confirm

### PR DESCRIPTION
**Pull request by eliseydah**
toasts and modals were added
image
image

### CODE REVIEW

File: frontend/src/components/articles/ArticleItem.vue [hunk 1]
```diff
@@ -17,19 +19,41 @@ const emit = defineEmits(["reloadArticleView"]);
...
+const updateArticleToast = () => {
+  toast.add({ severity: "info", summary: "Info", detail: "Article was updated", life: 3000 });
+};
 const onUpdateArticle = async () => {
   await updateArticle(`${editableArticle.value.id}`, editableArticle.value);
+  updateArticleToast();
 };
 
 const onDeleteArticle = async () => {
   await deleteArticle(`${editableArticle.value.id}`);
```
Comment (fr-ser) [inline frontend/src/components/articles/ArticleItem.vue:37]: Shouldn't we also add the confirmation dialog here?

File: frontend/src/components/articles/ArticleItem.vue
Comment (fr-ser) [inline frontend/src/components/articles/ArticleItem.vue:1]: Can we use "pi-save" as a save icon?

File: frontend/src/components/articles/ArticleItem.vue [hunk 2]
```diff
@@ -17,19 +19,41 @@ const emit = defineEmits(["reloadArticleView"]);
...
+    rejectLabel: "Cancel",
+    acceptLabel: "Delete",
+    accept: async () => {
+      await onDeleteArticle();
+      toast.add({
+        severity: "info",
+        summary: "Confirmed",
+        detail: "You have deleted this article",
+        life: 3000,
+      });
```
Comment (fr-ser) [inline frontend/src/components/articles/ArticleItem.vue:53]:
```suggestion
    message: "Wollen Sie den Artikel sicher wirklich löschen?",
    header: "Bestätigung",
    rejectLabel: "Abbrechen",
    acceptLabel: "Löschen",
    accept: async () => {
      await onDeleteArticle();
      toast.add({
        severity: "info",
        summary: "Bestätigung",
        detail: "Der Artikel wurde gelöscht.",
        life: 3000,
      });
```

File: frontend/src/components/articles/ArticleItem.vue [hunk 3]
```diff
@@ -17,19 +19,41 @@ const emit = defineEmits(["reloadArticleView"]);
...
 }>();
-
+const confirm = useConfirm();
+const toast = useToast();
 const articlesType = Object.values(ArticleKind);
 
 const editableArticle = ref(props.article);
 
+const updateArticleToast = () => {
+  toast.add({ severity: "info", summary: "Info", detail: "Article was updated", life: 3000 });
```
Comment (fr-ser) [inline frontend/src/components/articles/ArticleItem.vue:29]:
```suggestion
  toast.add({ severity: "info", summary: "Information", detail: "Die Änderungen wurden gespeichert", life: 3000 });
```

File: frontend/src/views/ArticleView.vue [hunk 4]
```diff
@@ -4,15 +4,26 @@ import InputText from "primevue/inputtext";
...
-//
+import { ArticleKind } from "@/global/types/appTypes";
 import type { Article } from "@/global/types/entities";
 
 async function reloadPage() {
   articlesList.value = (await getArticles()).data;
 }
 const articlesList = ref<Article[]>([]);
+
+// temp solution
```
Comment (fr-ser) [inline frontend/src/views/ArticleView.vue:17]: I would suggest such prefixes to be able to find them easily later.

```suggestion
// TODO: temp solution
```

File: frontend/src/views/ClientListView.vue [hunk 5]
```diff
@@ -18,6 +20,25 @@ async function removeClient(client: Client) {
...
+    header: "Confirmation",
+    rejectLabel: "Cancel",
+    acceptLabel: "Delete",
+    accept: async () => {
+      await removeClient(client);
+      toast.add({
+        severity: "info",
+        summary: "Confirmed",
+        detail: "You have deleted this client",
+        life: 3000,
```
Comment (fr-ser) [inline frontend/src/views/ClientListView.vue:37]:
```suggestion
    message: "Sind Sie sich sicher, dass der Kunde gelöscht werden soll?",
    header: "Bestätigung",
    rejectLabel: "Abbrechen",
    acceptLabel: "Löschen",
    accept: async () => {
      await removeClient(client);
      toast.add({
        severity: "info",
        summary: "Information",
        detail: "Der Kunde wurde gelöscht",
        life: 3000,
```

File: frontend/src/views/ClientView.vue [hunk 6]
```diff
@@ -6,47 +6,78 @@ import Dropdown from "primevue/dropdown";
...
+    header: "Confirmation",
+    rejectLabel: "Cancel",
+    acceptLabel: "Delete",
+    accept: async () => {
+      await onDeleteClient();
+      toast.add({
+        severity: "info",
+        summary: "Confirmed",
+        detail: "You have deleted this client",
+        life: 3000,
```
Comment (fr-ser) [inline frontend/src/views/ClientView.vue:38]: See here: https://github.com/fr-ser/scaffolding-management/pull/16/files#r1684544661

Maybe we can extract this method if we reuse it in two places? 🤔

File: frontend/src/views/ClientView.vue [hunk 7]
```diff
@@ -6,47 +6,78 @@ import Dropdown from "primevue/dropdown";
...
-
+const isEditing = computed(() => {
+  return Boolean(route.params.id);
+});
+const createClientToast = () => {
+  toast.add({ severity: "info", summary: "Info", detail: "Client was created", life: 3000 });
+};
+const updateClientToast = () => {
+  toast.add({ severity: "info", summary: "Info", detail: "Client was updated", life: 3000 });
+};
```
Comment (fr-ser) [inline frontend/src/views/ClientView.vue:62]:
```suggestion
const createClientToast = () => {
  toast.add({ severity: "info", summary: "Info", detail: "Ein neuer Kunde wurde erstellt", life: 3000 });
};
const updateClientToast = () => {
  toast.add({ severity: "info", summary: "Info", detail: "Die Änderung der Kundendaten wurde gespeichert", life: 3000 });
};
```

File: frontend/src/main.ts [hunk 8]
```diff
@@ -8,6 +10,8 @@ import Lara from "@/presets/lara";
 import router from "@/router";
 
 const app = createApp(App);
+app.use(ConfirmationService);
+app.use(ToastService);
```
Comment (fr-ser) [inline frontend/src/main.ts:14]: Do we have some configuration options? Can we make the Cancel button visually different? I.e. making it grey and to the left side?

Comment (eliseydah) [reply frontend/src/main.ts:14]: I changed the button color, but i have some problems with the position. Will customize the modal later.

File: frontend/src/views/ClientView.vue (hunk 6 above)
Comment (fr-ser) [reply frontend/src/views/ClientView.vue]: We have this code in two places, right? Can we extract it?

**Review by fr-ser — APPROVED** [review]
There is one comment left, but we can also fix that later 👌🏻

**Pull request merged**